platform/features/layout/src/LayoutController.js

```diff
@@ -127,6 +127,25 @@ define(
                 e.preventDefault();
             }
 
+            function getComposition(domainObject){
+                return domainObject.useCapability('composition');
+            }
+
+            function composeView (composition){
+                $scope.composition = composition;
+                return composition.map(function (object) {
+                        return object.getId();
+                    }) || [];
+            }
+
+            //Will fetch fully contextualized composed objects, and populate
+            // scope with them.
+            function refreshComposition() {
+                return getComposition($scope.domainObject)
+                    .then(composeView)
+                    .then(self.layoutPanels);
+            }
```

Comment:
refreshComposition could have ids available to it as an argument from the watch

I know, it's a bit icky. The reasoning behind this is actually tangentially related to the issue below. Basically I ignore the ids from the watch, because I subsequently fetch the composition from a capability (because it's no longer provided on `$scope`), and there is no guarantee the ids will match. So instead I just use the ids returned from the capability since that is the composition that is actually displayed in the layout. 

    There's some logic in mct-representation that prevents this for capabilities loaded via uses, so    this would be a regression.

Aha! I saw that bit of code and wondered what was going on there :) The comment said something about Angular reusing scope and subsequent representations, but I didn't realize it was relevant to this situation. 

It looks like the logic is relatively straightforward, but will need to be implemented carefully. I'll try to write a test scenario that verifies the fix as well.